src/ErrorHandler.php

```diff
@@ -14,20 +14,29 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Log\LoggerInterface;
 use Throwable;
 
 class ErrorHandler implements MiddlewareInterface
 {
     /** @var FormatterInterface[] */
     private $formatters = [];
 
+    /** @var LoggerInterface|null */
+    private $logger = null;
+
+    /** @var callable|null */
+    private $logCallback = null;
+
     /**
      * Configure the error formatters
      *
      * @param FormatterInterface[] $formatters
      */
-    public function __construct(?array $formatters = null)
+    public function __construct(?array $formatters = null, ?LoggerInterface $logger = null)
```

Comment:
I think the logger shouldn't be passed in the constructor because it's an optional feature. It should be a method (for example, `$middleware->logger(...)`).

Reply:
Dependencies belong in constructors... how else can we have dependency inversion?

Reply:
Yes, I agree. But in this case it's an optional feature. People may want to log the errors or not, so I see this as a configuration, not a core dependency of the middleware.

That being said, maybe you're right [in your comment](https://github.com/middlewares/error-handler/pull/17#issuecomment-2827126549) about the premise of this feature and it would be better to simply create a specific middleware for this.

@filisko ?

Reply:
A constructor parameter with a default value of `null` has no cost. With PHP named parameters, it doesn't even cost writing `null` to fill it in.